testng-core/src/test/java/test/configuration/issue2664/Issue2664Test.java

```diff
@@ -0,0 +1,138 @@
+package test.configuration.issue2664;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.Arrays;
+import java.util.List;
+import org.testng.Assert;
+import org.testng.TestNG;
+import org.testng.annotations.Test;
+import test.InvokedMethodNameListener;
+import test.SimpleBaseTest;
+
+public class Issue2664Test extends SimpleBaseTest {
+
+  @Test(description = "GITHUB-2663")
+  public void ensureThatConfigurationMethodsWithGroupDependencyWorks() {
+    List<String> expectedOrder1 =
+        Arrays.asList(
+            "beforeSuite3",
+            "beforeSuite1",
+            "beforeSuite2",
+            "beforeTest3",
+            "beforeTest1",
+            "beforeTest2",
+            "beforeClass3",
+            "beforeClass1",
+            "beforeClass2",
+            "beforeMethod3",
+            "beforeMethod1",
+            "beforeMethod2",
+            "test3",
+            "afterMethod3",
+            "afterMethod1",
+            "afterMethod2",
+            "beforeMethod3",
+            "beforeMethod1",
+            "beforeMethod2",
+            "test2",
+            "afterMethod3",
+            "afterMethod1",
+            "afterMethod2",
+            "beforeMethod3",
+            "beforeMethod1",
+            "beforeMethod2",
+            "test1",
+            "afterMethod3",
+            "afterMethod1",
+            "afterMethod2",
+            "afterClass3",
+            "afterClass1",
+            "afterClass2",
+            "afterTest3",
+            "afterTest1",
+            "afterTest2",
+            "afterSuite3",
+            "afterSuite1",
+            "afterSuite2");
+    TestNG testng = create(ConfigurationMethodWithGroupDependencySample.class);
+    InvokedMethodNameListener listener = new InvokedMethodNameListener();
+    testng.addListener(listener);
+    testng.run();
+    assertThat(testng.getStatus()).isEqualTo(0);
+    Assert.assertEquals(listener.getSucceedMethodNames(), expectedOrder1);
+  }
+
+  @Test(description = "GITHUB-2663")
+  public void ensureThatConfigurationMethodsWithGroupDependencyWorksWithBaseClass() {
+    ConfigurationMethodWithGroupDependencySample2.logs.clear();
```

Comment:
`logs` is not used in tests, you can remove it

Reply:
done